src/modules/app/containers/__tests__/NavigatorContainer.spec.js

```diff
@@ -0,0 +1,97 @@
+// @flow
+
+import TestRenderer from 'react-test-renderer'
+import { Provider } from 'react-redux'
+import Navigator from '../../components/Navigator'
+import configureMockStore from 'redux-mock-store'
+import { render } from '@testing-library/react-native'
+import * as React from 'react'
+import NavigatorContainer from '../NavigatorContainer'
+import waitForExpect from 'wait-for-expect'
+import AppSettings from '../../../settings/AppSettings'
+import AsyncStorage from '@react-native-community/async-storage'
+
+jest.mock('../../../../routes/landing/containers/LandingContainer', () => {
+  const Text = require('react-native').Text
+  return () => <Text>Landing</Text>
+})
+jest.mock('../../../../routes/dashboard/containers/DashboardContainer', () => {
+  const Text = require('react-native').Text
+  return () => <Text>Dashboard</Text>
+})
+jest.mock('../../../layout/containers/HeaderContainer', () => {
+  const Text = require('react-native').Text
+  return () => <Text>Header</Text>
+})
+jest.mock('@react-native-community/async-storage')
+jest.mock('react-native-share')
+
+const mockStore = configureMockStore()
+
+describe('NavigatorContainer', () => {
+  beforeEach(() => {
+    AsyncStorage.clear()
+    jest.resetModules()
+  })
+
+  it('should pass fetchCategory to Navigator', () => {
+    jest.doMock('../../components/Navigator', () => () => null)
+    const store = mockStore({})
+    const result = TestRenderer.create(<Provider store={store}><NavigatorContainer /></Provider>)
+    const navigator = result.root.findByType(Navigator)
```

Comment:
~This could be problematic here as the `Navigator` variable is not identical to the reference to the Navigator class used by the TestRenderer.~